device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala

```diff
@@ -65,21 +65,34 @@ class Routes(namespaceExtractor: Directive1[Namespace],
           messageBus.publish(DeviceCreated(ns, uuid, device.deviceName, device.deviceId, device.deviceType))
       }
 
-   onSuccess(f) { uuid =>
-     respondWithHeaders(List(Location(Uri("/devices/" + uuid.show)))) {
-        complete(Created -> uuid)
-     }
-   }
+    onSuccess(f) { uuid =>
+      respondWithHeaders(List(Location(Uri("/devices/" + uuid.show)))) {
+         complete(Created -> uuid)
+      }
+    }
   }
 
   def fetchSystemInfo(uuid: Uuid): Route =
-    complete(db.run(SystemInfo.findByUuid(uuid)))
-
-  def createSystemInfo(uuid: Uuid, data: Json): Route =
-    complete(Created -> db.run(SystemInfo.create(uuid, data)))
+    complete(db.run(SystemInfoRepository.findByUuid(uuid)))
+
+  def createSystemInfo(uuid: Uuid, data: Json): Route = {
+    val f = db.run(SystemInfoRepository.create(uuid, data))
+    f.onSuccess { case _ =>
+      val ns = db.run(DeviceRepository.deviceNamespace(uuid)).map { ns =>
+        GroupInfoRepository.list(ns).map { groupInfos =>
+          groupInfos.foreach { groupInfo =>
+            if (JsonComparison.getCommonJson(data, groupInfo.groupInfo).equals(groupInfo.groupInfo)) {
+              db.run(GroupMemberRepository.addGroupMember(groupInfo.id, uuid))
+            }
+          }
+        }
+      }
+    }
+    complete(Created -> f)
+  }
 
   def updateSystemInfo(uuid: Uuid, data: Json): Route =
-    complete(db.run(SystemInfo.update(uuid, data)))
+    complete(db.run(SystemInfoRepository.update(uuid, data)))
```

Comment:
This should maybe update which groups the device belongs to if the SystemInfo is new?